From 4eafe1a31e9a7f32435edf70ba8ddbc3f0ef692c Mon Sep 17 00:00:00 2001
From: Michel Alexandre Salim <salimma@fedoraproject.org>
Date: Thu, 6 Jul 2023 11:00:47 -0500
Subject: [PATCH] Fix to support fmt 10

In `fmt` 10 the error message is "invalid format specifier", not
"invalid type specifier". Use `.*` to capture both.

```
[ RUN      ] LoggerTest.formatError
/builddir/build/BUILD/folly-2023.07.03.00/folly/logging/test/LoggerTest.cpp:115: Failure
Value of: messages[0].first.getMessage()
Expected: matches regular expression "error formatting log message: .*invalid type specifier; format string: \"param1: \\{:06d\\}, param2: \\{:6.3f\\}\", arguments: 1234, hello world!"
  Actual: "error formatting log message: fmt::v10::format_error: invalid format specifier; format string: \"param1: {:06d}, param2: {:6.3f}\", arguments: 1234, hello world!"
[  FAILED  ] LoggerTest.formatError (0 ms)
```

Signed-off-by: Michel Alexandre Salim <salimma@fedoraproject.org>
---
 folly/logging/test/LoggerTest.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/folly/logging/test/LoggerTest.cpp b/folly/logging/test/LoggerTest.cpp
index be5849e82..c91ebf04f 100644
--- a/folly/logging/test/LoggerTest.cpp
+++ b/folly/logging/test/LoggerTest.cpp
@@ -115,7 +115,7 @@ TEST_F(LoggerTest, formatError) {
   EXPECT_THAT(
       messages[0].first.getMessage(),
       MatchesRegex(R"(error formatting log message: )"
-                   R"(.*invalid type specifier; )"
+                   R"(.*invalid .* specifier; )"
                    R"(format string: "param1: \{:06d\}, param2: \{:6.3f\}", )"
                    R"(arguments: 1234, hello world!)"));
   EXPECT_EQ("LoggerTest.cpp", pathBasename(messages[0].first.getFileName()));
-- 
2.41.0

